test/micro/org/openjdk/bench/java/lang/invoke/MethodHandlesTableSwitchRandom.java

```diff
@@ -0,0 +1,149 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.bench.java.lang.invoke;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.MutableCallSite;
+import java.util.Arrays;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+@BenchmarkMode(Mode.AverageTime)
+@Warmup(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 10, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@State(org.openjdk.jmh.annotations.Scope.Thread)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Fork(3)
+public class MethodHandlesTableSwitchRandom {
+
+    // Switch combinator test for a random input index, testing several switch sizes
+
+    private static final MethodType callType = MethodType.methodType(int.class, int.class);
+
+    private static final MutableCallSite cs = new MutableCallSite(callType);
+    private static final MethodHandle target = cs.dynamicInvoker();
+
+    private static final MethodHandle MH_SUBTRACT;
+    private static final MethodHandle MH_DEFAULT;
+    private static final MethodHandle MH_PAYLOAD;
+
+    static {
+        try {
+            MH_SUBTRACT = MethodHandles.lookup().findStatic(MethodHandlesTableSwitchRandom.class, "subtract",
+                    MethodType.methodType(int.class, int.class, int.class));
+            MH_DEFAULT = MethodHandles.lookup().findStatic(MethodHandlesTableSwitchRandom.class, "defaultCase",
+                    MethodType.methodType(int.class, int.class));
+            MH_PAYLOAD = MethodHandles.lookup().findStatic(MethodHandlesTableSwitchRandom.class, "payload",
+                    MethodType.methodType(int.class, int.class, int.class));
+        } catch (ReflectiveOperationException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+    }
+
+    // Using batch size since we really need a per-invocation setup
+    // but the measured code is too fast. Using JMH batch size doesn't work
+    // since there is no way to do a batch-level setup as well.
+    private static final int BATCH_SIZE = 1_000_000;
+
+    @Param({
+        "5",
+        "10",
+        "25",
+        "50",
+        "100"
+    })
```

Comment:
To reduce default runtime I think it'd be good to slim down the number of sizes we test here to 2 or 3 different ones.